csharp/tools/MauiModelTester/MainPage.xaml.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System.Diagnostics;
+
+namespace MauiModelTester;
+
+public partial class MainPage : ContentPage
+{
+    public MainPage()
+    {
+        InitializeComponent();
+
+        // See:
+        // ONNX Runtime Execution Providers: https://onnxruntime.ai/docs/execution-providers/
+        // Core ML: https://developer.apple.com/documentation/coreml
+        // NNAPI: https://developer.android.com/ndk/guides/neuralnetworks
+        ExecutionProviderOptions.Items.Add(nameof(ExecutionProviders.CPU));
+
+        if (DeviceInfo.Platform == DevicePlatform.Android)
+        {
+            ExecutionProviderOptions.Items.Add(nameof(ExecutionProviders.NNAPI));
+        }
+
+        if (DeviceInfo.Platform == DevicePlatform.iOS)
+        {
+            ExecutionProviderOptions.Items.Add(nameof(ExecutionProviders.CoreML));
+        }
+
+        // XNNPACK provides optimized CPU execution on ARM64 and ARM platforms for models using float
+        var arch = System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture;
+        if (arch == System.Runtime.InteropServices.Architecture.Arm64 ||
+            arch == System.Runtime.InteropServices.Architecture.Arm)
+        {
+            ExecutionProviderOptions.Items.Add(nameof(ExecutionProviders.XNNPACK));
+        }
+
+        ExecutionProviderOptions.SelectedIndex = 0; // default to CPU
+        ExecutionProviderOptions.SelectedIndexChanged += ExecutionProviderOptions_SelectedIndexChanged;
+
+        _currentExecutionProvider = ExecutionProviders.CPU;
+
+        // start creating session in background.
+        CreateInferenceSession();
+    }
+
+    private async Task CreateInferenceSession()
+    {
+        // wait if we're already creating an inference session.
+        if (_inferenceSessionCreationTask != null)
```

Comment:
it looks like different threads may have access to shared state (e.g., `_inferenceSessionCreationTask`). do we need to worry about races?

Reply:
Any UI action that would create a new inference session first disables the UI elements via SetBusy, so given that I think there should only ever be one thread from a user action that is calling CreateInferenceSession.

At startup we also call it, but _inferenceSessionCreationTask is null for that call.

Given that, is there a scenario where there would be a race?  

Another consideration is that as it's a test/example app I didn't want to implement unnecessary things that create noise

Reply:
I see a potential one between the call at startup and the call from the first UI action. it's probably not an issue in practice.